feature-libs/asm/customer-360/components/sections/asm-customer-coupon/asm-customer-coupon.component.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import {
+  Customer360Coupon,
+  Customer360CouponList,
+  Customer360Facade,
+  Customer360Type,
+} from '@spartacus/asm/customer-360/root';
+import { UserIdService } from '@spartacus/core';
+import {
+  BehaviorSubject,
+  Observable,
+  Subscription,
+  combineLatest,
+  of,
+} from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+import { ActiveCartFacade, CartVoucherFacade } from '@spartacus/cart/base/root';
+import { Customer360SectionContext } from '../customer-360-section-context.model';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'cx-asm-customer-coupon',
+  templateUrl: './asm-customer-coupon.component.html',
+})
+export class AsmCustomerCouponComponent implements OnInit, OnDestroy {
+  showErrorAlert$ = new BehaviorSubject<boolean>(false);
+  showErrorAlertForApplyAction$ = new BehaviorSubject<boolean>(false);
```

Comment:
Do we have chance to show two alert?

Reply:
Almost not. Only if apply 2 times at almost same time, it may show the alert.